plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -201,8 +201,12 @@ public void initModel(final Connection connection, final IProject project, boole
 				protected IStatus doRun(IProgressMonitor monitor) {
 					try {
 						List<IProject> projects = connection.getResources(ResourceKind.PROJECT);
+						//Project might be set by a master model that set connection to this slave model
+						//and thus caused loading projects job. While this job was in wait,
+						//master model might set project. It must be respected.
```

Comment:
We have 2 models which interact by one listening to the other.
Each property change sends a separate event.
We need to load resources in a separate job.
The job may be aware that while it loads resources for one of properties, some other property may be modified.

I thought about some alternative way, like setting connection+project to the slave model in one call. It seemed to me that changes to interface and implementation had to be too complicating, so that I decided it did not worth it.